### PR TITLE
[sources] add [sm] to source tree

### PR DESCRIPTION
**Pull request by jasonLaster**
Fixes Issue: #5663 

### Summary of Changes

This addresses the first of three issues with original sources that share a URL with a generated source. The other two issues are saved breakpoints and tabs. I think this is safe to land by itself and we can tackle the other two in their own PRs.

NOTE: the first commit is in another PR

screen shot 2018-08-08 at 5 57 06 pm

### CODE REVIEW

File: src/components/PrimaryPanes/SourcesTreeItem.js [hunk 1]
```diff
@@ -172,3 +178,23 @@ export default class SourceTreeItem extends Component<Props, State> {
...
+    return false;
+  }
+
+  const sources = getSourcesByURL(state, source.url);
+  return isOriginalId(source.id) && sources.length > 1;
+}
+
+const mapStateToProps = (state, props) => {
+  return {
+    isDup: hasMatchingGeneratedSource(state, props.source)
```
Comment (jasonLaster) [inline src/components/PrimaryPanes/SourcesTreeItem.js]: what's a good name for this property?

File: src/components/PrimaryPanes/SourcesTreeItem.js [hunk 2]
```diff
@@ -172,3 +178,23 @@ export default class SourceTreeItem extends Component<Props, State> {
...
+  };
+};
+
+export default connect(mapStateToProps, {
+  setProjectDirectoryRoot: actions.setProjectDirectoryRoot,
+  clearProjectDirectoryRoot: actions.clearProjectDirectoryRoot
+})(SourceTreeItem);
```
Comment (jasonLaster) [inline src/components/PrimaryPanes/SourcesTreeItem.js]: it's now connected so it is easy to check if it has a dup

File: src/components/PrimaryPanes/SourcesTreeItem.js (hunk 1 above)
Comment (darkwing) [reply src/components/PrimaryPanes/SourcesTreeItem.js]: `hasMatchingGeneratedSource` seems a reasonable property name to me :)

**Comment by darkwing** [comment]
We're still storing `prefs.debugger.tabs` as URLs (`["https://url1.com", ...]`) which is no longer viable if we're going to allow duplicate URLs, so I'm proposing we turn `prefs.debugger.tabs` into the following structure:

```js
[
    { url: "https://url1.com", generated: true }, 
    { url: "https://url1.com", generated: false }
]
```

This would allow duplicate tab URLs and maintain functionality of remembering tabs and connecting tabs to sources upon reload.  Thoughts?

**Comment by jasonLaster** [comment]
this is green, sans lint issue

**Comment by darkwing** [comment]
...and in turn, I think we'll need to use async storage (indexeddb) for tabs since someone could have many tabs open and we don't want to bloat prefs.

**Pull request merged**